examples/lottery-erc20/src/lib.rs

```diff
@@ -0,0 +1,418 @@
+mod random;
+mod lottery_info;
+
+use status::Status;
+use random::Random;
+use lottery_info::LotteryInfo;
+
+const PERCENTAGE_TOTAL: u16 = 100;
+const THIRTY_DAYS_IN_SECONDS: u64 = 60 * 60 * 24 * 30;
+
+#[elrond_wasm_derive::callable(Erc20Proxy)]
+pub trait Erc20 {
+    #[callback(transfer_from_callback)]
+    fn transferFrom(&self,
+        sender: &Address,
+        recipient: &Address,
+        amount: BigUint,
+        #[callback_arg] cb_lottery_name: Vec<u8>,
+        #[callback_arg] cb_sender: &Address);
+
+    #[callback(distribute_prizes_callback)]
+    fn transfer(&self, 
+        to: &Address, 
+        amount: BigUint, 
+        #[callback_arg] cb_lottery_name: Vec<u8>);
+}
+
+#[elrond_wasm_derive::contract(LotteryImpl)]
+pub trait Lottery {
+    
+    #[init]
+    fn init(&self, erc20_contract_address: Address) {
+        self.set_erc20_contract_address(&erc20_contract_address);
+    }
+
+    #[endpoint]
+    fn start(&self,
+        lottery_name: Vec<u8>,
+        ticket_price: BigUint, 
+        opt_total_tickets: Option<u32>, 
+        opt_deadline: Option<u64>,
+        opt_max_entries_per_user: Option<u32>,
+        opt_prize_distribution: Option<Vec<u8>>,
+        opt_whitelist: Option<Vec<Address>>) 
+        -> SCResult<()> {
+
+        self.start_lottery(lottery_name, ticket_price, opt_total_tickets, opt_deadline,
+            opt_max_entries_per_user, opt_prize_distribution, opt_whitelist)
+    }
+
+    #[endpoint(createLotteryPool)]
+    fn create_lottery_pool(&self,
+        lottery_name: Vec<u8>,
+        ticket_price: BigUint, 
+        opt_total_tickets: Option<u32>, 
+        opt_deadline: Option<u64>,
+        opt_max_entries_per_user: Option<u32>,
+        opt_prize_distribution: Option<Vec<u8>>,
+        opt_whitelist: Option<Vec<Address>>) 
+        -> SCResult<()> {
+
+        self.start_lottery(lottery_name, ticket_price, opt_total_tickets, opt_deadline,
+            opt_max_entries_per_user, opt_prize_distribution, opt_whitelist)
+    }
+    
+    fn start_lottery(&self,
+        lottery_name: Vec<u8>,
+        ticket_price: BigUint, 
+        opt_total_tickets: Option<u32>, 
+        opt_deadline: Option<u64>,
+        opt_max_entries_per_user: Option<u32>,
+        opt_prize_distribution: Option<Vec<u8>>,
+        opt_whitelist: Option<Vec<Address>>) 
+        -> SCResult<()> {
+
+        if lottery_name.is_empty() {
+            return sc_error!("Name can't be empty!");
+        }
+
+        let timestamp = self.get_block_timestamp();
+        
+        let total_tickets = opt_total_tickets.unwrap_or(u32::MAX);
+        let deadline = opt_deadline.unwrap_or(timestamp + THIRTY_DAYS_IN_SECONDS);
+        let max_entries_per_user = opt_max_entries_per_user.unwrap_or(u32::MAX);
+        let prize_distribution = opt_prize_distribution.unwrap_or([PERCENTAGE_TOTAL as u8].to_vec());
+        let whitelist = opt_whitelist.unwrap_or(Vec::new());
+
+        if self.status(lottery_name.clone()) != Status::Inactive {
```

Comment:
we prefer the following syntax:
require!(condition,
     "message");
it is more compact and easier to read